src/Guzzle/StatusDateOverrideMiddleware.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+namespace eLife\Journal\Guzzle;
+
+use eLife\ApiClient\MediaType;
+use InvalidArgumentException;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface;
+use function GuzzleHttp\json_decode;
+use function GuzzleHttp\json_encode;
+use function GuzzleHttp\Promise\promise_for;
+use function GuzzleHttp\Psr7\stream_for;
+
+final class StatusDateOverrideMiddleware
+{
+    private $authorizationChecker;
+    private $rdsArticles;
+
+    public function __construct(AuthorizationCheckerInterface $authorizationChecker, array $rdsArticles)
+    {
+        $this->authorizationChecker = $authorizationChecker;
+        $this->rdsArticles = $rdsArticles;
+    }
+
+    public function __invoke(callable $handler) : callable
+    {
+        if (empty($this->rdsArticles)) {
+            return $handler;
+        }
+
+        return function (RequestInterface $request, array $options = []) use (&$handler) {
+            return promise_for($handler($request, $options))->then(function (ResponseInterface $response) use ($request) {
+                try {
+                    $mediaType = MediaType::fromString($response->getHeaderLine('Content-Type'));
+                    $data = json_decode($response->getBody(), true);
+                } catch (InvalidArgumentException $e) {
+                    return $response;
```

Comment:
what case does this cover?

Reply:
It was introduced in `SubjectRewritingMiddleware` to handle non-JSON responses: https://github.com/elifesciences/journal/pull/870/commits/f67441364b75f4a4398f415e39c6fc56ad9f41d6

Reply:
Good to know, is it easy to introduce a unit test for it?

Reply:
@thewilkybarkid could you comment on whether this is necessary here. Why is it present in `SubjectRewritingMiddleware` but not the other middlewares?

Reply:
Can't remember if this gets attached to clients that access non-JSON (eg PDFs, images, supplementary files), but means that it bails out if it isn't which is safer behaviour.